frontend/src/scenes/surveys/surveyActivityDescriber.test.tsx

```diff
@@ -0,0 +1,260 @@
+        const result = describeFieldChange('response limit', 100, 200)
+        expect(getTextContent(result)).toBe('changed response limit from 100 to 200')
+    })
+
+    test('handles undefined before value', () => {
+        const result = describeFieldChange('iteration count', undefined, 5)
+        expect(getTextContent(result)).toBe('set iteration count to 5')
+    })
+
+    test('handles undefined after value', () => {
+        const result = describeFieldChange('iteration count', 5, undefined)
+        expect(getTextContent(result)).toBe('removed iteration count (was 5)')
+    })
+
+    test('handles empty string before value', () => {
+        const result = describeFieldChange('survey title', '', 'New Title')
+        expect(getTextContent(result)).toBe('set survey title to New Title')
+    })
+
+    test('handles empty string after value', () => {
+        const result = describeFieldChange('survey title', 'Old Title', '')
+        expect(getTextContent(result)).toBe('removed survey title (was Old Title)')
+    })
+
+    test('handles both values as empty strings', () => {
+        const result = describeFieldChange('survey title', '', '')
+        expect(getTextContent(result)).toBe('')
+    })
+
+    test('handles before and after as identical', () => {
+        const result = describeFieldChange('response limit', 100, 100)
+        expect(getTextContent(result)).toBe('')
+    })
+
+    test('handles string values with unit', () => {
+        const result = describeFieldChange('response time', 'fast', 'slow', 'seconds')
+        expect(getTextContent(result)).toBe('changed response time from fast seconds to slow seconds')
+    })
+
+    test('handles boolean values', () => {
+        const result = describeFieldChange('is active', false, true)
+        expect(getTextContent(result)).toBe('changed is active from false to true')
+    })
+
+    test('handles null values', () => {
+        const result = describeFieldChange('response limit', null, null)
+        expect(getTextContent(result)).toBe('')
+    })
+})
+
+describe('describeCommonChanges', () => {
+    const before: SurveyQuestion = {
+        question: 'What is your favorite color?',
+        description: 'Choose a color',
+        type: SurveyQuestionType.SingleChoice,
+        optional: false,
+        buttonText: 'Next',
+        choices: ['Red', 'Blue', 'Green'],
+    }
+    const after: SurveyQuestion = {
+        ...before,
+        question: 'What is your favorite animal?',
+        description: 'Choose an animal',
+        optional: true,
+        buttonText: 'Continue',
+    }
+
+    test('describes common changes', () => {
+        const changes = describeCommonChanges(before, after)
+        expect(changes).toHaveLength(4)
+        expect(getTextContent(changes[0])).toBe(
+            'changed question text from "What is your favorite color?" to "What is your favorite animal?"'
+        )
+        expect(getTextContent(changes[1])).toBe('updated description')
```

Comment:
why not also include the before/after description, like with the question change above?

Reply:
overlooked it; great feedback. Will add.